### PR TITLE
Feature/areas

### PR DESCRIPTION
**Pull request by vdubr**
- add support for non tiled stapialDataSources
- add asking for area relations in data endpoint
- add areaTreeLayerKey to the components/data requests
- add/fix some tests

### CODE REVIEW

File: src/state/Data/actions.js [hunk 1]
```diff
@@ -60,6 +65,84 @@ const getRestRelationsPages = (
...
+
+const checkSomeDSAreVector = spatialDataSources => {
+	const SDSvectorType = 'vector';
+	const someDSisVectorType = !_isEmpty(spatialDataSources)
+		? spatialDataSources.some(ds => ds.data?.type === SDSvectorType)
+		: false;
+	return someDSisVectorType;
+};
+
+const transformSDStoTypes = spatialDataSources => {
```
Comment (vlach1989) [inline src/state/Data/actions.js:86]: Why is the transformation needed?

**Review by vlach1989 — CHANGES_REQUESTED** [review]

File: src/state/Data/Components/selectors.js [hunk 2]
```diff
@@ -146,12 +146,25 @@ const getCommonFilterByComponentKey = createRecomputeSelector(componentKey => {
...
+
+		if (layerTemplateKey || modifiedLayerTemplateKey !== undefined) {
+			filter.layerTemplateKey = layerTemplateKey || modifiedLayerTemplateKey;
+		}
+
+		if (attributeKeys !== undefined) {
+			filter.attributeKeys = attributeKeys;
+		}
+
+		return filter;
```
[the remark this answers is not in the document]
Comment (vdubr) [reply src/state/Data/Components/selectors.js:167]: Did you made some steps in consolidations? If not, what do you suggest?

File: src/state/Data/actions.js (hunk 1 above)
Comment (vdubr) [reply src/state/Data/actions.js:86]: It creates objects just with key and type. These objects are used for determining if all or some of spatialDataSources are type of vector/raster etc.

So motivation for transformation is simplification of complicated objects.

File: src/state/AreaRelations/actions.js [hunk 3]
```diff
@@ -18,15 +18,50 @@ const ensureIndexed = (filter, order, start, length) =>
...
+const addIndex = common.addIndex(ActionTypes.AREA_RELATIONS);
 const useIndexedClearAll = common.useIndexedClearAll(
 	ActionTypes.AREA_RELATIONS
 );
 
+/**
+ * It ensure adding index and adding or updating received data from BE.
+ * Add relations to state only when areaRelations received, in case of empty areaRelations it adds only index.
+ * @param {Array} areaRelations Array received from BE contains areaRelations.
+ * @param {Object} filter Filler object contains modifiers, layerTemplateKey or areaTreeLevelKey and styleKey.
```
[the remark this answers is not in the document]
Comment (vdubr) [reply src/state/AreaRelations/actions.js:30]: True, removed: d9e81d914e349509572cd4300ebd1016bda67b28

File: src/state/AreaRelations/selectors.js [hunk 4]
```diff
@@ -4,69 +4,74 @@ import _ from 'lodash';
 import common from '../_common/selectors';
 
 const getSubstate = state => state.areaRelations;
-const getAll = common.getAll(getSubstate);
+// const getAll = common.getAll(getSubstate);
```
[the remark this answers is not in the document]
Comment (vdubr) [reply src/state/AreaRelations/selectors.js]: True, removed: d9e81d914e349509572cd4300ebd1016bda67b28

File: src/state/AreaRelations/selectors.js [hunk 5]
```diff
@@ -4,69 +4,74 @@ import _ from 'lodash';
...
+// /**
+//  * @return {Array|null}
+//  */
+// const getAllData = createSelector([getAll], relations => {
+// 	if (relations) {
+// 		return _.map(relations, relation => relation.data);
+// 	} else {
+// 		return null;
+// 	}
+// });
```
[the remark this answers is not in the document]
Comment (vdubr) [reply src/state/AreaRelations/selectors.js]: True, removed: d9e81d914e349509572cd4300ebd1016bda67b28

File: src/state/AreaRelations/selectors.js [hunk 6]
```diff
@@ -4,69 +4,74 @@ import _ from 'lodash';
...
+// 				}
+// 			});
+// 			return !_.isEmpty(filteredGroupedByLayerKey)
+// 				? filteredGroupedByLayerKey
+// 				: null;
+// 		} else {
+// 			return null;
+// 		}
+// 	}
+// )((state, layers) => JSON.stringify(layers));
```
[the remark this answers is not in the document]
Comment (vdubr) [reply src/state/AreaRelations/selectors.js]: True, removed: d9e81d914e349509572cd4300ebd1016bda67b28

File: src/state/Data/actions.js [hunk 7]
```diff
@@ -60,6 +65,84 @@ const getRestRelationsPages = (
 	}
 };
 
+// Check if all of spatialDataSources are tiled type.
+const checkAllDSAreTiled = spatialDataSources => {
+	const DSTILED = !_isEmpty(spatialDataSources)
```
[the remark this answers is not in the document]
Comment (vdubr) [reply src/state/Data/actions.js]: True, renamed: d9e81d914e349509572cd4300ebd1016bda67b28

File: src/state/Data/Components/selectors.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vlach1989) [reply src/state/Data/Components/selectors.js:167]: Not yet. However, I created the task at least :| Let's postpone it for now

**Review by vlach1989 — APPROVED** [review]

**Review by tmdmc — APPROVED** [review]

**Pull request merged**